src/gui/src/devsettings_widgets.cpp

```diff
@@ -0,0 +1,1255 @@
+  ck_bitset_dl->setEnabled(dev->features.supported & FEATURE_BITSETRDL);
+  ck_simulplus->setEnabled(dev->features.supported & FEATURE_SIMULPLUS);
+
+  ck_prec->setChecked(dev->features.enabled & FEATURE_POWEREC);
+  ck_hcdr->setChecked(dev->features.enabled & FEATURE_HIDECDR);
+  ck_ss->setChecked(dev->features.enabled & FEATURE_SINGLESESSION);
+  ck_spdrd->setChecked(dev->features.enabled & FEATURE_SPEEDREAD);
+  ck_bitset->setChecked(dev->features.enabled & FEATURE_BITSETR);
+  ck_bitset_dl->setChecked(dev->features.enabled & FEATURE_BITSETRDL);
+  ck_simulplus->setChecked(dev->features.enabled & FEATURE_SIMULPLUS);
+
+  connect(ck_prec, SIGNAL(toggled(bool)), this, SLOT(set_prec(bool)));
+  connect(ck_hcdr, SIGNAL(toggled(bool)), this, SLOT(set_hcdr(bool)));
+  connect(ck_ss, SIGNAL(toggled(bool)), this, SLOT(set_ss(bool)));
+  connect(ck_spdrd, SIGNAL(toggled(bool)), this, SLOT(set_spdrd(bool)));
+  connect(ck_bitset, SIGNAL(toggled(bool)), this, SLOT(set_bitset(bool)));
+  connect(ck_bitset_dl, SIGNAL(toggled(bool)), this, SLOT(set_bitset_dl(bool)));
+  connect(ck_simulplus, SIGNAL(toggled(bool)), this, SLOT(set_simulplus(bool)));
+}
+
+void devctlCommon::set_prec(bool en) { dev->setFeature(FEATURE_POWEREC, en); }
+void devctlCommon::set_hcdr(bool en) { dev->setFeature(FEATURE_HIDECDR, en); }
+void devctlCommon::set_ss(bool en) {
+  dev->setFeature(FEATURE_SINGLESESSION, en);
+}
+void devctlCommon::set_spdrd(bool en) {
+  dev->setFeature(FEATURE_SPEEDREAD, en);
+}
+void devctlCommon::set_bitset(bool en) { dev->setFeature(FEATURE_BITSETR, en); }
+void devctlCommon::set_bitset_dl(bool en) {
+  dev->setFeature(FEATURE_BITSETRDL, en);
+}
+void devctlCommon::set_simulplus(bool en) {
+  dev->setFeature(FEATURE_SIMULPLUS, en);
+}
+
+// GigaRec control
+
+devctlGigarec::devctlGigarec(device *idev, QWidget *p)
+    : QGroupBox("GigaRec", p) {
+  QString gc;
+
+  dev = idev;
+  layout = new QVBoxLayout(this);
+  layout->setMargin(3);
+  layout->setSpacing(3);
+
+  setCheckable(true);
+  bool ratio_14 = dev->dev.startsWith("CD-R   PREMIUM");
+  bool ratio_11 = dev->dev.startsWith("DVDR   PX-755") ||
+                  dev->dev.startsWith("DVDR   PX-760") ||
+                  dev->dev.startsWith("CD-R   PREMIUM2");
+
+  grec_ratio = new TextSlider(Qt::Vertical, this);
+  grec_ratio->setTickPosition(QSlider::TicksBelow);
+  grec_ratio->setUpsideDown(1);
+
+  gc = QString::number((int)(dev->media.ctots * 0.6) >> 9) + " MB";
+  grec_ratio->addItem("0.6", ratio_14);
+  gc = QString::number((int)(dev->media.ctots * 0.7) >> 9) + " MB";
+  grec_ratio->addItem("0.7", 1);
+  gc = QString::number((int)(dev->media.ctots * 0.8) >> 9) + " MB";
+  grec_ratio->addItem("0.8", 1);
+  gc = QString::number((int)(dev->media.ctots * 0.9) >> 9) + " MB";
+  grec_ratio->addItem("0.9", ratio_11);
+  // grec_ratio->addItem("OFF",1);
+  grec_ratio->addItem("", 0);
+  gc = QString::number((int)(dev->media.ctots * 1.1) >> 9) + " MB";
```

Comment:
Codacy found an issue: [Variable 'gc' is reassigned a value before the old one has been used.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)